storage/host-flash/host-flash.c

```diff
@@ -15,155 +15,161 @@
 #include <stdint.h>
 #include <stdio.h>
 #include <unistd.h>
+#include <string.h>
 
 #include "host-flash.h"
 
 static struct {
-    int filefd;
-    size_t flashsz;
-    size_t sectorsz;
+	int filefd;
+	size_t flashsz;
+	size_t sectorsz;
 } hostflash_common;
 
 ssize_t hostflash_read(unsigned int addr, void *buff, size_t bufflen)
 {
-    ssize_t stat;
-    int read = 0;
+	ssize_t stat;
+	int readsz = 0;
 
-    if (addr + bufflen > hostflash_common.flashsz)
-        return -EINVAL;
+	if (addr + bufflen > hostflash_common.flashsz)
+		return -EINVAL;
 
-    while ((stat = pread(hostflash_common.filefd, (void *)((char *)buff + read), bufflen - read, addr + read)) != 0) {
-        if (stat < 0) {
-            if (errno == -EINTR)
-                continue;
+	while ((stat = pread(hostflash_common.filefd, (void *)((char *)buff + readsz), bufflen - readsz, addr + readsz)) != 0) {
+		if (stat < 0) {
+			if (errno == EINTR)
+				continue;
 
-            return stat;
-        }
+			return stat;
+		}
 
-        read += stat;
-        if (read == bufflen)
-            break;
-    }
+		readsz += stat;
+		if (readsz == bufflen)
+			break;
+	}
 
-    return read;
+	return readsz;
 }
 
 
 ssize_t hostflash_write(unsigned int addr, void *buff, size_t bufflen)
 {
-    char currentByte = 0;
-    size_t wrote = 0;
-    ssize_t stat;
+	char tempTab[256];
+	size_t wrote = 0, i;
+	int len;
+	ssize_t stat, readsz;
+
+	if (addr + bufflen > hostflash_common.flashsz)
+		return -EINVAL;
+
+	while (wrote != bufflen) {
+		if ((bufflen - wrote) >= sizeof(tempTab))
+			len = sizeof(tempTab);
+		else
+			len = (bufflen - wrote) % sizeof(tempTab);
```

Comment:
Is this modulo needed? 
bufflen - wrote < sizeof(tempTab)  => (bufflen - wrote) % sizeof(tempTab) === bufflen - wrote
Am I right?

Reply:
Right, modulo isn't needed.

Reply:
Done